test: formatting skip messages for TAP parsing

thefourtheye opened the pull request:
This patch makes the skip messages consistent so that the TAP plugin
in CI can parse the messages properly. The format will be

    1..0 # Skipped: [Actual reason why the test is skipped]

cc @bnoordhuis

jbergstroem commented:
Nice. LGTM.

Fishrock123 commented:
Perhaps we should inform the tests if they are being run under tap so this can be configured in the common test helpers file per test run output type?

thefourtheye commented:
@fishrock123 You mean like a common ffunction with the template string? How can we know if the tap is enabled?

jbergstroem commented:
I think the text is as readable in tap output as in plain text. The test runner knows what mode we're in, but it feels like an unnecessary complication. We should really start abstracting output if we care about this.

Fishrock123 commented:
> I think the text is as readable in tap output as in plain text. The test runner knows what mode we're in, but it feels like an unnecessary complication. We should really start abstracting output if we care about this.

Eh, ok. LGTM

bnoordhuis commented:
LGTM.  Maybe replace the process.exit calls with return statements while you're there, that should make reasoning about the test suite's remaining process.exit calls easier.

thefourtheye commented:
@bnoordhuis Done! PTAL :-)

bnoordhuis commented:
LGTM.  CI: https://jenkins-iojs.nodesource.com/view/iojs/job/iojs+any-pr+multi/120/

thefourtheye commented:
@bnoordhuis I have never seen TAP in action. But looking at the Extended TAP Results in the CI, the changes didn't make any difference I guess. What am I missing?

jbergstroem commented:
@thefourtheye you can see the output in the raw stdout logs. Not sure if jenkins registers skipped test as a part of the parsed output list.

thefourtheye commented:
@jbergstroem Hmmm, in Console output also, I couldn't see any logs, but only the TAP results (ok/not ok)

jbergstroem commented:
It looks like the test runner actually returns 'ok' on skip (well, return code 0). I'll have a look at this. This patch is still good to go though.

Fishrock123 commented:
New CI: https://jenkins-iojs.nodesource.com/view/iojs/job/iojs+any-pr+multi/123/

thefourtheye commented:
Okay, I rebased `master`, pulled in @jbergstroem's https://github.com/nodejs/io.js/pull/2129 and changed the `process.exit` to `return` and squashed the commits.

jbergstroem commented:
I'd suggest we keep the PR's separate. I probably have to update #2129 shortly (regex change) and neither PR relies on the other to work. No strong opinion about it though.

thefourtheye commented:
@jbergstroem https://github.com/nodejs/io.js/pull/2129 is not the `test.py` changes, but the "missing crypto changes" which you landed sometime earlier ;-)

jbergstroem commented:
@thefourtheye ah, I'll just continue studying the art of not reading properly.

thefourtheye commented:
@jbergstroem lol, no problem :-) Do we need a separate CI run for this?

jbergstroem commented:
@thefourtheye might as well; they're just idling. https://jenkins-iojs.nodesource.com/job/iojs+any-pr+multi/126/

bnoordhuis commented:
LGTM with a comment.

thefourtheye commented on test/sequential/test-setproctitle.js:
```diff
@@ -3,8 +3,8 @@
 if ('linux freebsd darwin'.indexOf(process.platform) === -1) {
-  console.error('Skipping test, platform not supported.');
-  process.exit();
+  console.log(`1..0 # Skipped: Unsupported platform [${process.platform}]`);
```
@bnoordhuis PTAL :-)

brendanashworth commented:
@thefourtheye is this landable?

thefourtheye commented:
@brendanashworth We can land this anytime but the relevant changes are in #2130. Cc @jbergstroem

jbergstroem commented:
I'm LGTM for landing this. #2130 just makes it "visible".

thefourtheye commented:
Thanks people. Landed in 79c865a53feef1828d8c93e953acbeb01ee2cb74 and 69298d36cfe1d5ad2ade0e64586959cc18c7ea4e
